### PR TITLE
store/tikv: support debug PB in client.

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What problem does this PR solve? 
A sub PR for fast analyze. Reference: #9973 

### What is changed and how it works?
Just add `DebugPB.Client` in `tikv.client`.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes

 - Need to be included in the release note

### CODE REVIEW

File: store/tikv/tikvrpc/tikvrpc.go [hunk 1]
```diff
@@ -147,6 +188,20 @@ type Request struct {
 	MvccGetByKey       *kvrpcpb.MvccGetByKeyRequest
 	MvccGetByStartTs   *kvrpcpb.MvccGetByStartTsRequest
 	SplitRegion        *kvrpcpb.SplitRegionRequest
+
+	DebugGet                    *debugpb.GetRequest
```
Comment (lamxTyler) [inline store/tikv/tikvrpc/tikvrpc.go]: What about remove these unused fields? This struct is already too large. Refer https://github.com/pingcap/tidb/issues/9641

Comment (lzmhhh123) [reply store/tikv/tikvrpc/tikvrpc.go]: Okay.

Comment (lzmhhh123) [reply store/tikv/tikvrpc/tikvrpc.go]: Have been removed.

File: store/tikv/tikvrpc/tikvrpc.go [hunk 2]
```diff
@@ -61,6 +62,20 @@ const (
 	CmdMvccGetByKey CmdType = 1024 + iota
 	CmdMvccGetByStartTs
 	CmdSplitRegion
+
+	CmdDebugGet CmdType = 2048 + iota
```
Comment (lamxTyler) [inline store/tikv/tikvrpc/tikvrpc.go]: Do we need this? I guess the order does not matter here.

File: store/tikv/tikvrpc/tikvrpc.go [hunk 3]
```diff
@@ -592,6 +651,23 @@ func CallRPC(ctx context.Context, client tikvpb.TikvClient, req *Request) (*Resp
...
+	resp.Type = req.Type
+	var err error
+	switch req.Type {
+	case CmdDebugGetRegionProperties:
+		resp.DebugGetRegionProperties, err = client.GetRegionProperties(ctx, req.DebugGetRegionProperties)
+	default:
+		return nil, errors.Errorf("invalid request type: %v", req.Type)
+	}
+	if err != nil {
+		return nil, errors.Trace(err)
```
Comment (lamxTyler) [inline store/tikv/tikvrpc/tikvrpc.go]: No need to trace the error.

File: store/tikv/tikvrpc/tikvrpc.go (hunk 2 above)
Comment (lzmhhh123) [reply store/tikv/tikvrpc/tikvrpc.go]: Yeah, but it's clear to keep order here, also no other impaction.

Comment (lamxTyler) [reply store/tikv/tikvrpc/tikvrpc.go]: `String` and `IsDebugReq` will be called for every request, it is better to reduce the useless work.

Comment (lzmhhh123) [reply store/tikv/tikvrpc/tikvrpc.go]: All right.

File: store/tikv/client.go [hunk 4]
```diff
@@ -579,6 +580,17 @@ func (c *rpcClient) SendRequest(ctx context.Context, addr string, req *tikvrpc.R
 		}
 	}
 
+	if req.IsDebugReq() {
+		client := debugpb.NewDebugClient(connArray.Get())
+		ctx1, cancel := context.WithTimeout(ctx, timeout)
+		defer cancel()
+		resp, err := tikvrpc.CallDebugRPC(ctx1, client, req)
+		if err != nil {
+			return nil, errors.Trace(err)
```
Comment (zz-jason) [inline store/tikv/client.go]: no need to trace this error

File: store/tikv/tikvrpc/tikvrpc.go [hunk 5]
```diff
@@ -592,6 +610,23 @@ func CallRPC(ctx context.Context, client tikvpb.TikvClient, req *Request) (*Resp
 	return resp, nil
 }
 
+// CallDebugRPC launches a debug rpc call.
+func CallDebugRPC(ctx context.Context, client debugpb.DebugClient, req *Request) (*Response, error) {
+	resp := &Response{}
```
Comment (zz-jason) [inline store/tikv/tikvrpc/tikvrpc.go]: how about:
```go
resp := &Response{Type:req.Type}
```

File: store/tikv/client.go [hunk 6]
```diff
@@ -579,6 +580,17 @@ func (c *rpcClient) SendRequest(ctx context.Context, addr string, req *tikvrpc.R
 		}
 	}
 
+	if req.IsDebugReq() {
+		client := debugpb.NewDebugClient(connArray.Get())
+		ctx1, cancel := context.WithTimeout(ctx, timeout)
+		defer cancel()
+		resp, err := tikvrpc.CallDebugRPC(ctx1, client, req)
```
Comment (zz-jason) [inline store/tikv/client.go]: can be simplified to:
```go
return tikvrpc.CallDebugRPC(ctx1, client, req)
```

File: store/tikv/tikvrpc/tikvrpc.go [hunk 7]
```diff
@@ -592,6 +610,23 @@ func CallRPC(ctx context.Context, client tikvpb.TikvClient, req *Request) (*Resp
...
+	resp := &Response{Type: req.Type}
+	resp.Type = req.Type
+	var err error
+	switch req.Type {
+	case CmdDebugGetRegionProperties:
+		resp.DebugGetRegionProperties, err = client.GetRegionProperties(ctx, req.DebugGetRegionProperties)
+	default:
+		return nil, errors.Errorf("invalid request type: %v", req.Type)
+	}
+	if err != nil {
```
Comment (zz-jason) [inline store/tikv/tikvrpc/tikvrpc.go]: ditto

**Pull request merged**